### DISCUSSION TITLE
German/Deutsch Number Format in Switzerland

### DISCUSSION THREAD

**Discussion by chgoeldi** [home-assistant/frontend#20342] [Localization] [answered]
Hi

When the language setting is "Deutsch", the UI allows to choose between the number formats:

1,234,567.89
1.234.567,89
1 234 567,89

However, in Switzerland, we usually use **1'234'567.89**. Could someone please add this option?

Best Regards,

**Comment by NoRi2909** [comment]
@chgoeldi That is in fact a bug (or two bugs) you should report:

In Settings > System > General you can set your country to 'Switzerland' which should set the 'system locale' to use apostrophes for the thousands separator and a period as the decimal separator.

Then in your user settings you could select "Use system locale" for the number format, while still keeping 'Deutsch', 'Français', 'Italiano' etc. as your UI language.

I note that we even have 'Schwiizerdütsch' as a language option in the User settings but that does not switch to 1'234'567.89 either. So there is a second, identical bug there.

**Reply by michaelherger** [reply]
Did you ever follow up on this? I happen to have opened an issue earlier today: https://github.com/home-assistant/frontend/issues/27254.

I would agree that automatic evaluation should not only use the language, but the locale's country into account, too. 

@piitaya mentioned we might need a new locale `de-CH`. Would that mean a new translation? I believe the German translation would be more or less identical. It's more about the localisation of numbers (and probably other entities) than the language. Is there a way to have a new locale, but have it inherit all the textual content from another language? Or would it be preferable to have better locale evaluation to take the language _and_ the country into account in code, eg. in `numberFormatToLocale()`:

https://github.com/home-assistant/frontend/blob/4cedfffb71542c35000acad5faba254b7d9059d8/src/common/number/format_number.ts#L25-L40

I believe the code could concatenate the country to the language (at least for some languages like `de`, `pt`, `en`, `fr`), falling back to the language alone if needed?

**Reply by piitaya** [reply]
Yeah it can be a solution. But that means we will have to update all the `formatNumber` to now pass `hass.config.country` Also, the country is not configurable in the same screen (it's under system and not user). So I personally would not mix those.
i think it's easier to add `quote_decimal` option so user can configure it.

**Reply by michaelherger** [reply, marked as answer]
https://github.com/home-assistant/frontend/pull/27268
